lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -813,6 +815,15 @@ namespace {
         return finish(true, TC.resolveDeclRefExpr(unresolved, DC));
       }
 
+      if (auto selector = dyn_cast<ObjCSelectorExpr>(expr)) {
```

Comment:
Can you leave a short comment explaining why selector references are exempt from circularity checks.